[#41133] Improve documentation for configuration variables to include…

adam-op opened the pull request:
… allowed values

[#41133] Improve documentation for configuration variables to include allowed values

https://community.openproject.org/work_packages/41133

Commit:
[#41133] Improve documentation for configuration variables to include allowed values

[#41133] Improve documentation for configuration variables to include allowed values

https://community.openproject.org/work_packages/41133

machisuji commented on docs/installation-and-operations/configuration/environment/README.md:
```diff
@@ -36,18 +36,18 @@ For example, given the following configuration:
 You can override it by defining the following environment variable:
 
-    OPENPROJECT_STORAGE_TMP__PATH=/some/other/path
+    STORAGE_TMP__PATH=/some/other/path
```
Removing the OPENPROJECT here is wrong. Not using the `OPENPROJECT` prefix is deprecated too.
Same with the `STORAGE_TYPE` below in line 44. So please keep the prefix here.

adam-op replied:
HI @machisuji  
I was just thinking and did not find a conclusion
PS: for the configuration variables I might have a meeting with @cbliard this week.

machisuji reviewed (changes requested):
Looks good to me until we have a rake task or UI in OpenProject to list all the settings.

Please just amend the one remake I made and then this can be merged.

Commit:
[#41133] Improve documentation for configuration variables to include allowed values

[#41133] Improve documentation for configuration variables to include allowed values

https://community.openproject.org/work_packages/41133

added OPENPROJECT_ to STORAGE_TMP__PATH and STORAGE_TYPE

machisuji commented on docs/installation-and-operations/configuration/environment/README.md:
```diff
@@ -61,79 +61,109 @@ Please note: The Configuration is a HashWithIndifferentAccess and thus it should
+OPENPROJECT_SMTP__USER__NAME: ""
+OPENPROJECT_SQL__SLOW__QUERY__THRESHOLD: 2000
+OPENPROJECT_STATSD: {"host"=>nil, "port"=>8125}
+OPENPROJECT_WEB: {"workers"=>2, "timeout"=>120, "wait_timeout"=>10, "min_threads"=>4, "max_threads"=>16}
```
@adam-op Ah one more thing, could you please "expand" these nested values? I.e. don't give the hash as a default but list the individual values into separate variables. E.g. not just OPENPROJECT_WEB with the hash as the default but `OPENPROJECT_WEB_WORKERS: 2`, `OPENPROJECT_WEB_TIMEOUT: 120` etc..

Same with OPENPROJECT_STATSD.

adam-op replied:
i will wait until the new rake task is implemented
